discretisedfield/io/vtk.py

```diff
@@ -64,13 +64,17 @@ def _from_vtk(cls, filename):
             # Old writing routine did write to points instead of cells.
             return cls._from_vtk_legacy(filename)
 
+        valid_idx = None
+
         vdims = []
         for i in range(cell_data.GetNumberOfArrays()):
             name = cell_data.GetArrayName(i)
             if name == "field":
                 field_idx = i
-            elif name.endswith("-component"):
-                vdims.append(name[: -len("-component")])
+            elif name == "valid":
+                valid_idx = i
+            elif name not in ["field", "norm"]:
```

Comment:
```suggestion
            elif name not in ["norm"]:
```
This should be sufficient? `"field"` is already tested in the first `if`.